src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas2/StatusUpdateServiceTest.kt

```diff
@@ -429,4 +445,309 @@ class StatusUpdateServiceTest {
+                ) &&
+                  it.data.eventDetails.newStatus == Cas2Status(
+                  name = "offerDeclined",
+                  label = "Offer declined or withdrawn",
+                  description = "The accommodation offered has been declined or withdrawn.",
+                  statusDetails = listOf(
+                    Cas2StatusDetail("exampleStatusDetail", ""),
+                  ),
+                )
+              },
+            )
+          }
+        }
+
+        @Test
+        fun `alerts Sentry when the Referrer does not have an email`() {
+          val submittedApplicationWithNoReferrerEmail = Cas2ApplicationEntityFactory()
+            .withCreatedByUser(NomisUserEntityFactory().withEmail(null).produce())
+            .withCrn("CRN123")
+            .withNomsNumber("NOMSABC")
+            .withSubmittedAt(OffsetDateTime.now().randomDateTimeBefore(2))
+            .produce()
+
+          val assessmentWithNoEmail = Cas2AssessmentEntityFactory()
+            .withApplication(submittedApplicationWithNoReferrerEmail).produce()
+
+          every { mockAssessmentRepository.findByIdOrNull(assessmentWithNoEmail.id) } answers
+            {
+              assessmentWithNoEmail
+            }
+
+          mockkStatic(Sentry::class)
+
+          every {
+            Sentry.captureException(
+              RuntimeException(
+                "Email not found for User ${submittedApplicationWithNoReferrerEmail.createdByUser.id}. " +
+                  "Unable to send email when updating status of Application ${submittedApplicationWithNoReferrerEmail.id}",
+                NotFoundException("Email not found for User ${submittedApplicationWithNoReferrerEmail.createdByUser.id}"),
+              ),
+            )
+          } returns SentryId.EMPTY_ID
+
+          statusUpdateService.createForAssessment(
+            assessmentId = assessmentWithNoEmail.id,
+            statusUpdate = applicationStatusUpdateWithDetail,
+            assessor = assessor,
+          )
+
+          verify(exactly = 1) {
+            Sentry.captureException(
+              any(),
+            )
+          }
+        }
+      }
+
+      @Nested
+      inner class WhenUnsuccessful {
+
+        @BeforeEach
+        fun setUp() {
+          every { mockAssessmentRepository.findByIdOrNull(assessment.id) } answers
+            {
+              null
+            }
+        }
+
+        @Test
+        fun `does NOT ask the domain event service to create a status-updated event`() {
+          statusUpdateService.createForAssessment(
+            assessmentId = assessment.id,
+            statusUpdate = applicationStatusUpdate,
+            assessor = assessor,
+          )
+
+          verify(exactly = 0) {
+            mockDomainEventService.saveCas2ApplicationStatusUpdatedDomainEvent(any())
+          }
+
+          verify(exactly = 0) { mockEmailNotificationService.sendEmail(any(), any(), any()) }
+        }
+      }
+    }
+  }
```

Comment:
I can see the test coverage for the new `CreateForAssessment` is the same as `Create` for applications - excellent.